cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -836,8 +836,18 @@ set(Boost_ADDITIONAL_VERSIONS
 # Thrift needs Boost if we're building the bundled version with version < 0.13,
 # so we first need to determine whether we're building it
 if(ARROW_WITH_THRIFT AND Thrift_SOURCE STREQUAL "AUTO")
+  find_package(PkgConfig QUIET)
```

Comment:
I think you added this @thisisnic ? Did it resolve the issue you thought it was going to? And do you remember what that error was?

Reply:
This should be added just before `pkg_check_modules(...)`.

Reply:
@jonkeane It is, and it fixes something, though as @kou says, it should be moved